driver/src/price_estimation/dexag/api.rs

```diff
@@ -0,0 +1,183 @@
+#![allow(dead_code)]
+
+use anyhow::{anyhow, Context, Result};
+use ethcontract::Address;
+use isahc::prelude::*;
+use serde::{de::Error as _, Deserialize, Deserializer};
+use serde_with::rust::display_fromstr;
+use std::time::Duration;
+use url::Url;
+
+#[cfg_attr(test, mockall::automock)]
+/// Parts of the dex.ag api.
+pub trait DexagApi {
+    /// https://docs.dex.ag/api/tokens
+    fn token_list_full(&self) -> Result<Vec<Token>>;
+    /// https://docs.dex.ag/api/price
+    fn price(
+        &self,
+        from: &str,
```

Comment:
Shouldn't from/to be of type Token?

Reply:
`Token` in this module is the result of the `token_list` call. For this function dex.ag uses token symbols which is just one part of the struct Token so I don't want to require filling a full Token struct. In my mind this module is trying to be a thin wrapper around the http api so I am happy this interface.

Reply:
But we are only going to request the price for tokens that have been previously returned by `token_list_full`, aren't we? 
Thus I find it more logical to have the types be the same (gives the reader a clue that this output is used as an input here). We could also change `token_list_full` to return &str but I think having strong types here is better.